shopinvader_api_address/routers/address_service.py

```diff
@@ -0,0 +1,113 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import authenticated_partner
+from odoo.addons.shopinvader_schema_address.schemas import (
+    BillingAddress,
+    ShippingAddress,
+)
+
+from ..schemas import BillingAddressUpdate,ShippingAddressCreate,ShippingAddressUpdate
+
+# create a router
+address_router = APIRouter(tags=["addresses"])
+
+# --- Billing addresses ---
+
+
+@address_router.get("/addresses/billing", response_model=BillingAddress)
+def get_billing_address(
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> BillingAddress:
+    """
+    Get billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    address = partner._get_shopinvader_billing_address()
+    return BillingAddress.from_res_partner(address)
+
+
+@address_router.post(
+    "/addresses/billing", response_model=BillingAddress, status_code=201
+)
+def update_billing_address(
+    data: BillingAddressUpdate,
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> BillingAddress:
+    """
+    Update billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    vals = data.to_res_partner_vals()
+    # sudo() is needed because some addons override the write
+    # function of res.partner to do some checks before writing.
+    # These checks need more rights than what we are giving to
+    # the enspoint's user
+    # (e.g. snailmail/models/res_partner.py)
+    address = partner.sudo()._update_shopinvader_billing_address(vals)
+    return BillingAddress.from_res_partner(address)
+
+
+# --- Shipping address ---
+
+
+@address_router.get("/addresses/shipping", response_model=List[ShippingAddress])
+@address_router.get(
+    "/addresses/shipping/{address_id}", response_model=List[ShippingAddress]
+)
+def get_shipping_address(
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+    address_id: int | None = None,
+) -> List[ShippingAddress]:
+    """
+    Get shipping addresses of authenticated user
+    Can be used to get every shipping address: /addresses/shipping
+    Can be used to get one specific address: /addresses/shipping/address_id
+    """
+    addresses = partner._get_shopinvader_shipping_addresses(address_id)
+    return [ShippingAddress.from_res_partner(rec) for rec in addresses]
+
+
+@address_router.post(
+    "/addresses/shipping", response_model=ShippingAddress, status_code=201
+)
+@address_router.post("/addresses/shipping/{address_id}", response_model=ShippingAddress)
+def create_update_shipping_address(
+    data: ShippingAddressCreate | ShippingAddressUpdate,
```

Comment:
Hm, I think you need to do two functions. Otherwise you'd need to document in the swagger and implement a check that ShippingAddressCreate is for create and ShippingAddressCreate is for update.

Reply:
Yes, I realised this but I already pushed (the split is now done)